Reactive messaging, kafka and native: No ReactiveStreamEngine implementation found

A native binary for a project using reactive messaging fails with the error below.

Project's source code: https://github.com/michalszynkiewicz/preview-search-engine

```
2019-05-15 23:03:03,438 INFO  [io.sma.rea.mes.ext.MediatorManager] (main) Connecting emitter to sink search-terms
java.lang.RuntimeException: java.lang.IllegalStateException: No ReactiveStreamsEngine implementation found!
        at io.quarkus.smallrye.reactivemessaging.runtime.SmallRyeReactiveMessagingLifecycle.onApplicationStart(SmallRyeReactiveMessagingLifecycle.java:20)
        at io.quarkus.smallrye.reactivemessaging.runtime.SmallRyeReactiveMessagingLifecycle_Observer_onApplicationStart_4e8937813d9e8faff65c3c07f88fa96615b70e70.notify(Unknown Source)
        at io.quarkus.arc.EventImpl$Notifier.notify(EventImpl.java:244)
        at io.quarkus.arc.EventImpl.fire(EventImpl.java:85)
        at io.quarkus.arc.runtime.LifecycleEventRunner.fireStartupEvent(LifecycleEventRunner.java:39)
        at io.quarkus.arc.runtime.ArcDeploymentTemplate.handleLifecycleEvents(ArcDeploymentTemplate.java:115)
        at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent21.deploy(Unknown Source)
        at io.quarkus.runner.ApplicationImpl1.doStart(Unknown Source)
        at io.quarkus.runtime.Application.start(Application.java:100)
        at io.quarkus.runtime.Application.run(Application.java:212)
        at io.quarkus.runner.GeneratedMain.main(Unknown Source)
Caused by: java.lang.IllegalStateException: No ReactiveStreamsEngine implementation found!
        at org.eclipse.microprofile.reactive.streams.operators.core.ReactiveStreamsEngineResolver.instance(ReactiveStreamsEngineResolver.java:64)
        at org.eclipse.microprofile.reactive.streams.operators.core.CompletionRunnerImpl.run(CompletionRunnerImpl.java:37)
        at io.smallrye.reactive.messaging.extension.MediatorManager.weaving(MediatorManager.java:233)
        at io.smallrye.reactive.messaging.extension.MediatorManager.initializeAndRun(MediatorManager.java:149)
        at io.smallrye.reactive.messaging.extension.MediatorManager_ClientProxy.initializeAndRun(Unknown Source)
        at io.quarkus.smallrye.reactivemessaging.runtime.SmallRyeReactiveMessagingLifecycle.onApplicationStart(SmallRyeReactiveMessagingLifecycle.java:18)
        ... 10 more
2019-05-15 23:03:03,449 ERROR [io.qua.run.StartupContext] (main) Running a shutdown task failed: java.util.concurrent.RejectedExecutionException: Task io.vertx.core.impl.TaskQueue$$Lambda$76ad1f053e61859eedb59b3e5ac366287a291713@7fb330e4b9a8 rejected from java.util.concurrent.ThreadPoolExecutor@7fb367226980[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 0]
        at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2063)
        at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:830)
        at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1379)
        at io.vertx.core.impl.TaskQueue.execute(TaskQueue.java:93)
        at io.vertx.core.impl.ContextImpl.executeBlocking(ContextImpl.java:288)
        at io.vertx.core.impl.ContextImpl.executeBlocking(ContextImpl.java:242)
        at io.vertx.core.impl.EventLoopContext.executeBlocking(EventLoopContext.java:23)
        at io.vertx.core.impl.ContextImpl.executeBlocking(ContextImpl.java:247)
        at io.vertx.core.impl.EventLoopContext.executeBlocking(EventLoopContext.java:23)
        at io.vertx.kafka.client.producer.impl.KafkaWriteStreamImpl.close(KafkaWriteStreamImpl.java:209)
        at io.vertx.kafka.client.producer.impl.KafkaWriteStreamImpl.close(KafkaWriteStreamImpl.java:204)
        at io.vertx.kafka.client.producer.impl.KafkaWriteStreamImpl.close(KafkaWriteStreamImpl.java:199)
        at io.smallrye.reactive.messaging.kafka.KafkaSink.close(KafkaSink.java:115)
        at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:891)
        at io.smallrye.reactive.messaging.kafka.KafkaMessagingProvider.terminate(KafkaMessagingProvider.java:30)
        at io.smallrye.reactive.messaging.kafka.KafkaMessagingProvider_Observer_terminate_01ca61423b802db2b2ec45ffe53148a5efa33e27.notify(Unknown Source)
        at io.quarkus.arc.EventImpl$Notifier.notify(EventImpl.java:244)
        at io.quarkus.arc.EventImpl$Notifier.notify(EventImpl.java:234)
        at io.quarkus.arc.ArcContainerImpl.shutdown(ArcContainerImpl.java:262)
        at io.quarkus.arc.Arc.shutdown(Arc.java:53)
        at io.quarkus.arc.runtime.ArcDeploymentTemplate$1.run(ArcDeploymentTemplate.java:53)
        at io.quarkus.runtime.StartupContext.close(StartupContext.java:59)
        at io.quarkus.runner.ApplicationImpl1.doStart(Unknown Source)
        at io.quarkus.runtime.Application.start(Application.java:100)
        at io.quarkus.runtime.Application.run(Application.java:212)
        at io.quarkus.runner.GeneratedMain.main(Unknown Source)

Exception in thread "main" java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl1.doStart(Unknown Source)
        at io.quarkus.runtime.Application.start(Application.java:100)
        at io.quarkus.runtime.Application.run(Application.java:212)
        at io.quarkus.runner.GeneratedMain.main(Unknown Source)
Caused by: java.lang.RuntimeException: java.lang.IllegalStateException: No ReactiveStreamsEngine implementation found!
        at io.quarkus.smallrye.reactivemessaging.runtime.SmallRyeReactiveMessagingLifecycle.onApplicationStart(SmallRyeReactiveMessagingLifecycle.java:20)
        at io.quarkus.smallrye.reactivemessaging.runtime.SmallRyeReactiveMessagingLifecycle_Observer_onApplicationStart_4e8937813d9e8faff65c3c07f88fa96615b70e70.notify(Unknown Source)
        at io.quarkus.arc.EventImpl$Notifier.notify(EventImpl.java:244)
        at io.quarkus.arc.EventImpl.fire(EventImpl.java:85)
        at io.quarkus.arc.runtime.LifecycleEventRunner.fireStartupEvent(LifecycleEventRunner.java:39)
        at io.quarkus.arc.runtime.ArcDeploymentTemplate.handleLifecycleEvents(ArcDeploymentTemplate.java:115)
        at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent21.deploy(Unknown Source)
        ... 4 more
Caused by: java.lang.IllegalStateException: No ReactiveStreamsEngine implementation found!
        at org.eclipse.microprofile.reactive.streams.operators.core.ReactiveStreamsEngineResolver.instance(ReactiveStreamsEngineResolver.java:64)
        at org.eclipse.microprofile.reactive.streams.operators.core.CompletionRunnerImpl.run(CompletionRunnerImpl.java:37)
        at io.smallrye.reactive.messaging.extension.MediatorManager.weaving(MediatorManager.java:233)
        at io.smallrye.reactive.messaging.extension.MediatorManager.initializeAndRun(MediatorManager.java:149)
        at io.smallrye.reactive.messaging.extension.MediatorManager_ClientProxy.initializeAndRun(Unknown Source)
        at io.quarkus.smallrye.reactivemessaging.runtime.SmallRyeReactiveMessagingLifecycle.onApplicationStart(SmallRyeReactiveMessagingLifecycle.java:18)
        ... 10 more

```

--- a/extensions/smallrye-reactive-messaging-kafka/deployment/pom.xml
+++ b/extensions/smallrye-reactive-messaging-kafka/deployment/pom.xml
@@ -24,6 +24,10 @@
         </dependency>
         <dependency>
             <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-smallrye-reactive-streams-operators-deployment</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
             <artifactId>quarkus-kafka-client-deployment</artifactId>
         </dependency>
         <dependency>

